### ISSUE TITLE
App store screenshots thread

### ISSUE THREAD

**Issue by offset-torque** [koreader/koreader#12027]
I will start making the screenshots @pazos . I think a temporary issue like this is the best option for tracking and feedback. I will upload the screenshots to a temporary directory next to the guide directory. I will ping you as I upload the screenshots and you can give feedback for them.

List below will act as a to-do list and I will update it as we go. 

**Legend will be:**
- 🔴 for problem marker
- 🟡 for "waiting for your feedback"
- 🟢 for "ready to publish"

 so we can track the progress.

## Phone (1080x1920)
- Reading screen with highlight types and night mode - Split
- Reading screen CJK split
- Reading screen Arabic, Hebrew split
- Bottom menu items - Combined
- Dictionary/Wikipedia/Translate menu screenshot
- Book map
- Page browser
## Generic (1404x1872)
- Reading screen with highlight types and night mode - Split
- Reading screen CJK split
- Reading screen Arabic, Hebrew split
- Bottom menu items - Combined
- Dictionary/Wikipedia/Translate menu screenshot
- Book map
- Page browser
## Desktop (2000x1400)
**REMINDER: We can't use split screen screenshots here because they prohibit edits**
- Reading screen
- Reading screen night mode
- Reading screen Chinese **OR** Japanese **OR** Korean
- Reading screen Arabic **AND/OR** Hebrew
- Bottom menu items - One or two relevant
- Dictionary/Wikipedia/Translate menu screenshot
- Book map
- Page browser

Firstly I need you and @Frenzie to have a look at the list above and decide which items to add/remove.
Especially multi-lingual ones are not decided yet (and remember the Hebrew religious text discussion from the other thread).

**Comment by offset-torque** [comment]
Here is the first screenshot: **Reading screen+Highlight types+Night mode split**
Please comment on anything that needs to change like font related things, status bar, margins etc.

![Phone_ReadingScreen_Highlights_NightMode_Split](https://github.com/koreader/koreader/assets/84420582/5391daf2-a465-4a94-ba3c-887535de425a)

**Comment by Frenzie** [comment]
As for something that looks really nice on screenshots the statistics calendar is also a good candidate. The only problem is filling it with data. I agree with your choices; for screenshots you want visual things that elucidate rather than merely duplicate that which is already in the text.

> (and remember the Hebrew religious text discussion from the other thread).

The Tanach/Pentateuch/Bible is simply cultural heritage imho, not some religious stance. Incidentally the Plantijn Polyglot Bible is simply a joy to see in person. But text suggestions are better left for people who actually read the language like @yparitcher. If you don't like Genesis you could always take the Song of Songs for example, although surely there has to be a decent literary Yiddish text rendered in Hebrew or some such, relating to Tevye the Milkman for example (see [here](https://www.yiddishbookcenter.org/collections/yiddish-books/spb-nybc201649/sholem-aleichem-tevye-der-milkhiker) but that's a PDF).

<https://en.wikipedia.org/wiki/Plantin_Polyglot>
![image](https://github.com/koreader/koreader/assets/202757/e7874206-4d65-4574-8ecd-754bcdd7c006)

I'd be worried we don't render it nearly that nicely though. ;-)

**Comment by pazos** [comment]
> I will upload the screenshots to a temporary directory next to the guide directory. I will ping you as I upload the screenshots and you can give feedback for them.

As you will. You could also make a PR against https://github.com/koreader/koreader-artwork. Images can be easily reviewed as part of the PR.

> which items to add/remove.

As far as I'm concerned I'm going to be fine with whatever set of screenshots you agree upon. Just remember we need a caption per screenshot type.

**Comment by pazos** [comment]
@offset-torque: any news?

We don't need too fancy screens. Just an improvement over https://flathub.org/apps/rocks.koreader.KOReader and captions :)

Let me know if I could help somehow :)

**Comment by offset-torque** [comment]
I was planning to do the screenshots after dealing with the current guide update but release keeps getting postponed it seems. So I will start them now, no need to delay this more. Thanks for the help offer pazos, I can manage this. I will seek your and Frenzie's feedback here as I make them. Can you clarify the "captions". They won't be on the images according to the guidelines, right? Will it be filename or just an associated screenshot description table (filename -> caption).

**Comment by Frenzie** [comment]
The final option makes the most sense.

**Comment by pazos** [comment]
Yay, captions are chunks of text that give some context or explain the image you're seeing.

They're only used on flathub, AFAICT.

A program without captions: https://flathub.org/apps/rocks.koreader.KOReader
A program with captions that don't add too much info: https://flathub.org/apps/dev.geopjr.Calligraphy
A program with "working" captions: https://flathub.org/apps/org.kde.kdenlive

Most of the programs with working captions describe the view you're seeing on the image.

**Comment by offset-torque** [comment]
Small status update. Just after I wrote my last message, we got the new release. Currently I am finishing the user guide update. After that I will deal with the screenshots.

**Comment by offset-torque** [comment]
Ok, guide update is done so it is time to deal with the screenshots. I noticed that we can directly work from my Github pages just like the guide update feedback so I made this file. 

There are some questions in yellow marker. First step is deciding the screenshot types. Please have a look. I will also add the completed ones to this file so we will have a tidy overview of progress.

https://offset-torque.github.io/screenshots.html

**Comment by poire-z** [comment]
I don't think we need all phone/generic/desktop screenshots variations.
Phone and generic have the same A/R, so let's have just one with the dpi & font set to how it looks for most of us on such phone or eink tablets.
And I think we don't need any desktop landscape screenshot: this is not our target audience, and most stuff is not that optimized (because we mostly don't care) for lanscape, ie. the often too wide dialog and even the top menu).

As for the topics:
- colored highlight: no opinion
- CJK split: I guess you mean Chinese & Japanese & Korean typography ? I'd say: don't advertisze that, these users will see by themselves. No xenophobia from me (afterall, I made these typography enhancements), but I'm not for so much advertizing to Chinese: we most often get report in Chinese where people make no effort (and people who can make the effort will find/know about KOReader features). Also, it looks like there are devices makers/sellers in China that do install KOReader, often an older version - and we are not there to ensure their customer support (and answer questions which may be obsolete, and inform about how to update to a newer version, etc...)
- Hebrew/Persian/Arabic split: there, I would have no such problem with showing bits of text for all (for typography, some epub page with prepared paragraphs in each languages could be fine). For the RTL UI, I dunno.
- Book map / Page browsers: I think what we have is quite enough as far as showing the capabilities. If you are talking about screenshots without legend text on them, I'm off-topic :)
- Font tab weight/hinting/kerning: what you have in the user guide is quite enough, people will know and play with it if curious. We don't need to show screenshots of these.

Sorry if I'm off topic and I have not given the answers you expected :)

**Comment by yparitcher** [comment]
@offset-torque 
re: hebrew/persian/arabic, you can show a page with a paragraph if each of these languages, which should not be offensive.

as @Frenzie  mentioned above, the statistics calendar is a good candidate for a screenshot.

@poire-z 
These screenshots are for the app stores / website for new users (whom are browsing and definitely are not  looking to the user guide yet) to give a quick taste/overview of KOReader's features/strengths at a glance.

**Comment by pazos** [comment]
> I don't think we need all phone/generic/desktop screenshots variations.

We don't **NEED** them but that's what we have:

- generic: https://koreader.rocks/
- phone: https://f-droid.org/en/packages/org.koreader.launcher.fdroid/
- desktop: https://flathub.org/apps/rocks.koreader.KOReader

> Phone and generic have the same A/R, so let's have just one with the dpi & font set to how it looks for most of us on such phone or eink tablets.

That's fine for me :)

> And I think we don't need any desktop landscape screenshot: this is not our target audience, and most stuff is not that optimized (because we mostly don't care) for lanscape, ie. the often too wide dialog and even the top menu).

We don't need them (landscape). It is just for aesthetics (it looks better on the web page an any client that lets you install the package). In order to follow flathub guidelines we just need them to be 1.) with window decorations 2.) WYSIWYG (ie: not edited pictures, like https://github.com/koreader/koreader/issues/12027#issuecomment-2167308161)

**Comment by offset-torque** [comment]
> Sorry if I'm off topic and I have not given the answers you expected :)

I have no expectations, only volunteered to save someone else from doing this :)
It is totally open discussion currently, you (devs) decide, I will make them.

> Phone and generic have the same A/R, so let's have just one with the dpi & font set to how it looks for most of us on such phone or eink tablets.

But they don't have the same aspect ratio. Phone is 9:16, generic is more like our e-ink devices which is 3:4. Did I misunderstood this sentence?

**Comment by poire-z** [comment]
> But they don't have the same aspect ratio. Phone is 9:16, generic is more like our e-ink devices which is 3:4. Did I misunderstood this sentence?

I meant they are _nearly_ the same A/R - that is: portrait :)
Again, phones in 9/16 are probably not the target audience and not the best A/R to show off KOReader. It looks best on 3/4 eInk devices (the A/R of a book) - and I guess Android eInk devices (the target audience among Android) are such 3/4. So, it feels we don't need to show 9/16 - and let users with 9/16 phones see by themselves how bad it is for reading.

**Comment by pazos** [comment]
I'm not sure what's the fuss with "target audience". 


The target audience for f-droid are f-droid users and the target audience for flathub are flathub users. Granted that neither of those groups are for whom the application was developed in the first place.

But since we're there we should provide honest screenshots (we already do!). Those screenshots and the description act as a filter for lurking users on random phones searching for "pdf reader", while KOReader users can see the program they want to install on whatever devices they want.

I don't think that making the app available on platforms other than **tier 1** (kobo/kindle, maybe pocketbook nowadays) makes the project harder to maintain but if that's the general feeling we should avoid to publish the program on those third party stores.

I use KOReader on a chromebook, on my laptop, on my tablet and my ereader. I don't care at all about people saying it doesn't fit in whatever framework used by other applications of that platform. I do care about the easy to install & to update is on each platform where I would like to run the program.

**Comment by Frenzie** [comment]
> if that's the general feeling we should avoid to publish the program on those third party stores.

Nah, I think the program works great in 9/16 and 9/20 on a phone, also for many a PDF. For comics I prefer a different program though, with free drag & zoom.

I'm not super enthused about Flathub right now but that's mostly because I don't trust the auto-update to work when tagging a release soon. Providing a few screenshots that are 90% "generic" anyway isn't an issue.

> But they don't have the same aspect ratio. Phone is 9:16, generic is more like our e-ink devices which is 3:4. Did I misunderstood this sentence?

They should remain separate as they are. :-) "Generic" is the Kobo Aura One resolution, which I purposefully used as a stand-in for (nearly) all ereaders. Phones are a lot more stretched. I assume 2000x1400 is the current size of screenshots on Flathub? In any case that or something like it is also good for that purpose.

----

* colored highlights: I opt for color. I think it's rather self-evident that it'll be grayscale on your average ereader.

* kerning etc.: doesn't hurt, I'd put it on best when taking the screenshot :-)

**Comment by poire-z** [comment]
> But since we're there we should provide honest screenshots (we already do!). 
> I do care about the easy to install & to update is on each platform where I would like to run the program.

I'm of course fine with that.

> I don't think that making the app available on platforms other than **tier 1** makes the project harder to maintain but if that's the general feeling we should avoid to publish the program on those third party stores.

I'm not asking to avoid that.

I would just like (sharing my feeling, not even asking) that we don't do too much (and especially good) marketing. (Or use this project as an exercice for marketing, advetisement, popularity gaining, etc...)
We gain nothing by gaining more users (except _complications_)), not even quality feedback (this, we get it from the users that took the pain of finding the software).
It's already hard to follow everthing (the amount of posts on reddit has risen recently), at some point it could become overwhelming.

**Comment by pazos** [comment]
> We gain nothing by gaining more users (except complications)), not even quality feedback (this, we get it from the users that took the pain of finding the software).

Agree. But f-droid and flathub are not that good to discover applications and we don't get a lot of low effort tickets related to them.

The numbers of downloads for those stores **together** don't outperform  the number of github downloads for the same platform on a single release :)

That's because KO isn't particulary attractive for those platform users and the few who try, without previous experience, desist after a few minutes trying to grasp the program :p 

But there's somebody out there that's actually a KOReader user and want a simple way to get it installed on regular computers ;)

> It's already hard to follow everthing (the amount of posts on reddit has risen recently), at some point it could become overwhelming

Indeed. I'm in reddit. It doesn't worth the time to give any kind of support there (I've tried a couple of times) :p 

But as far as the github project it shouldn't be overwhelming as long as it isn't materialized on low effort tickets.

In that regard I would suggest:

- request **english** as the language for communication
- request **reading the documentation** the first time you submit a FR, bug report or PR.
- request **searching** before opening to avoid duplicates

for bug reports:

- document how to report bugs
- request reproducing the bug on latest nightly before submit a new ticket

for feature request:

- limit the scope of the feature request we're interested here (99'9% can be done outside this organization) and gh issues is not a hub to keep track of people wishes :)
- redirect people with other ideas to either gh discussions or the forum or the chat.


As this is a "community" repo I think nobody enforced any rules to interact with this repo/org and that works fine for the most part. Now, if any kind of popularity bring the "community" here more work, we might be better setting a few thingies that we agree upon and enforce the rules.


Back to the topic: If we're agaisnt good marketing and nice screenshots indeed pushing screenshots with ratios where KO doesn't shine seems a good idea to me :)

**Comment by offset-torque** [comment]
> pazos - But since we're there we should provide honest screenshots (we already do!). Those screenshots and the description act as a filter for lurking users on random phones searching for "pdf reader", while KOReader users can see the program they want to install on whatever devices they want.

Yep this is how I think too. Let me explain my reasoning so it may clarify the discussion. Screenshots proposed and their message:

- **Reading screen with highlight types:** This is how our interface looks like and we support many types of highlights so it is also good for academic use. 
- **CJK and/or RTL screenshot(s):** We support other languages too.
- **Font settings screenshot:** You can customize the text as you wish.
- **Book map and/or page browser:** We have this unique and useful features (for more serious reading) that other apps don't have. (Personally just the Book Map would be enough for me to try KOReader if I see it in the screenshots on F-Droid)
- **Reading statistics (proposed by **yparitcher**):** If you are a "serious" reader you can track your progress.

I think problem we are trying to solve is that e-reader applications started to become a part of "note taking, to-do, calculator" group which every new programmer tests their skills on. Majority of these reader apps are very simple. Very basic highlights, no export, very limited text customization (sometimes just font size), supports only English... So our screenshots can be a filter against these low-effort ones (Although F-Droid is much better in this regard than the Play Store).

> pazos - As this is a "community" repo I think nobody enforced any rules to interact with this repo/org and that works fine for the most part. Now, if any kind of popularity bring the "community" here more work, we might be better setting a few thingies that we agree upon and enforce the rules.

I think the rules **pazos** mentioned are good guidelines. I generally see a very similar structure on more popular oss repos. Without precautions, especially toxic/entitled users become a big burden on a project once it reaches a certain size. We didn't hit that milestone yet thankfully.

>  pazos - Indeed. I'm in reddit. It doesn't worth the time to give any kind of support there (I've tried a couple of times) :p

Reddit demographic changed very significantly as it became more popular. Even specialized subreddits like e-ink, e-reader etc. are nearly useless now. No interesting discussions, no insightful comments. Everything is "lol lmao fr fr". Probably this is the demographic we should really avoid, thankfully they don't read long things :))
